### PR TITLE
Feature/43 explosion damage logic

### PR DESCRIPTION
**Pull request by RyanJST**
Basic form of explosion damage, including a raycast to test to see if there is a clear line of sight.  If not, there is no damage dealt.

### CODE REVIEW

File: Assets/Scripts/Damage/ExplosionDamage.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using UnityEngine;
+
+public class ExplosionDamage : MonoBehaviour
+{
+    /// <summary>
+    /// Struct that contains what the fire damage is supposed to be.
```
Comment (overseer25) [inline Assets/Scripts/Damage/ExplosionDamage.cs:6]: Copy pasted comment?

File: Assets/Scripts/Damage/ExplosionDamage.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    /// </summary>
+    DamageInfo explosionDamage;
+
+    /// <summary>
+    /// Method called before first frame update
+    /// </summary>
+    public void Start()
+    {
+        explosionDamage.DamageAmount = 50;
```
Comment (overseer25) [inline Assets/Scripts/Damage/ExplosionDamage.cs:15]: I think the DamageInfo values should be configurable in the inspector.

File: Assets/Scripts/Damage/ExplosionDamage.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        RaycastHit rayHitObject;
+        IKillable item = other.gameObject.GetComponent<IKillable>();
+
+        if (item != null)
+        {
+            Debug.Log("Explosion Damage Field Entered");
+            if (Physics.Raycast(transform.position, other.transform.position - transform.position, out rayHitObject))
+            {
+                bool colliderExposed = rayHitObject.collider == other;
```
Comment (overseer25) [inline Assets/Scripts/Damage/ExplosionDamage.cs:29]: Nit-pick, but I'd like the conditional to be surrounded in parentheses.

Comment (RyanJST) [reply Assets/Scripts/Damage/ExplosionDamage.cs]: Just put in paraentheses

File: Assets/Scripts/Damage/ExplosionDamage.cs (hunk 2 above)
Comment (RyanJST) [reply Assets/Scripts/Damage/ExplosionDamage.cs]: Set them up to be serial fields.  Also updated the prefab to the defaults that I had.

File: Assets/Scripts/Damage/ExplosionDamage.cs (hunk 1 above)
Comment (RyanJST) [reply Assets/Scripts/Damage/ExplosionDamage.cs]: Changed fire to explosion.

**Pull request merged**